### PR TITLE
Implement error::Error and fmt::Display for string::ParseError

### LINKED ISSUE #31106: std::string::ParseError doesn't implement neither std::error::Error nor std::fmt::Display.
subj

### PR DESCRIPTION
**Pull request by dirk**
Fixes #31106.
##### Couple of questions:
- [x] I wasn't sure of the correct `#[stable(...)]` definition to use here. Happy to fix it if it's incorrect.
- [x] `ParseError` is sort of an ephemeral non-error, but do let me know if the implementation of `error::Error` for it should return something more descriptive than "parse error".

### CODE REVIEW

File: src/libstd/error.rs [hunk 1]
```diff
@@ -182,6 +182,13 @@ impl Error for string::FromUtf16Error {
     }
 }
 
+#[stable(feature = "str_parse_error2", since = "1.8.0")]
+impl Error for string::ParseError {
+    fn description(&self) -> &str {
+        "parse error"
```
Comment (alexcrichton) [inline src/libstd/error.rs]: Ah here you can probably do the same `match *self {}` trick to avoid the need to return anything here at all.

**Comment by alexcrichton** [comment]
Looks good to me, thanks @dirk! The stability attributes are fine here (impls are insta-stable) and we picking any old name for the feature is fine

File: src/libstd/error.rs (hunk 1 above)
Comment (dirk) [inline src/libstd/error.rs]: @alexcrichton: Done in https://github.com/dirk/rust/commit/365deeb19a2cff74e3d215622b1ab411581b99e9!

**Comment by alexcrichton** [comment]
@bors: r+ 365deeb19a2cff74e3d215622b1ab411581b99e9

Thanks @dirk!

**Comment by dirk** [comment]
@alexcrichton: Thanks, glad to help out! :smile:

**Pull request merged**